peps/pep-0736.rst

```diff
@@ -0,0 +1,361 @@
+
+* This objection is is subjective and many community members disagree.
+* A nearly-identical syntax is already established for f-strings.
+* Programmers will, as ever, adjust over time.
+
+The feature is confusing
+------------------------
+
+We argue that:
+
+* Introducing new features typically has this impact temporarily.
+* The syntax is very similar to the established ``f'{x=}'`` syntax.
+* The feature and syntax are familiar from other popular modern languages.
+* The expansion of ``x=`` to ``x=x`` is in fact a trivial feature and inherently
+  significantly less complex than ``*arg`` and ``**kwarg`` expansion.
+* This particular syntactic form has been independently proposed on numerous
+  occasions, indicating that it is the most obvious [8]_ [9]_ [13]_.
+
+The feature is not explicit
+---------------------------
+
+This is based on a misunderstanding of the Zen of Python. Keyword arguments are
+fundamentally more explicit than positional ones where argument assignment is
+only visible at the function definition. On the contrary, the proposed syntactic
+sugar contains all the information as is conveyed by the established keyword
+argument syntax but without the redundancy. Moreover, the introduction of this
+syntactic sugar incentivises use of keyword arguments, making typical Python
+codebases more explicit.
+
+The feature adds another way of doing things
+--------------------------------------------
+
+The same argument can be made against all syntax changes. This is a simple
+syntactic sugar, much as ``x += 1`` is sugar for ``x = x + 1`` when ``x`` is an
+integer. This isn't tantamount to a 'new way' of passing arguments but a more
+readable notation for the same way.
+
+Renaming the variable in the calling context will break the code
+----------------------------------------------------------------
+
+A ``NameError`` would make the mistake abundantly clear. Moreover, text editors
+could highlight this based on static analysis ‒ ``f(x=)`` is exactly equivalent
+to writing ``f(x=x)``. If ``x`` does not exist, modern editors have no problem
+highlighting the issue.
+
+Recommendations
+===============
+
+As with any other language feature, the programmer should exercise their own
+judgement about whether to use it in any given context. We do not recommend
+enforcing a rule to use the feature in all cases where it may be applicable.
+
+Reference Implementation
+========================
+
+`A proposed implementation <https://github.com/Hels15/cpython/tree/last-build>`_
+for cpython has been provided by @Hels15.
+
+References
+==========
+
+.. [1] Issue 36817: Add = to f-strings for easier debugging. - Python tracker
+   https://bugs.python.org/issue36817
+.. [2] Ruby keyword argument syntactic sugar
+   https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/#:~:text=Other%20Notable%20New%20Features
+.. [3] ReasonML named argument punning
+   https://reasonml.github.io/docs/en/function#:~:text=Named%20argument%20punning
```

Comment:
```suggestion
   https://reasonml.github.io/docs/en/function
```

Reply:
Again this was intentional. Happy to remove if there's a hard preference but I think the highlights add value.

Reply:
Will switch to #function-application for Firefox support.